frontend/test/cypress/support/commands.ts

```diff
@@ -23,4 +23,22 @@
 //
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+/// <reference path="../support/index.d.ts" />
 import '@testing-library/cypress/add-commands';
+
+Cypress.Commands.add('isWithinViewport', { prevSubject: true }, (element: any) => {
+  const windowInnerWidth = Cypress.config(`viewportWidth`);
+  const windowInnerHeight = Cypress.config(`viewportHeight`);
+
+  const bounding = element[0].getBoundingClientRect();
```

Comment:
Why `element[0]` instead of `element`?

Reply:
This `element` here was actually a jQuery object. 

From [here](https://stackoverflow.com/a/6974603), found the explanation for **`[0]`**: 
> $("selector")[0] // Accesses the first DOM element in this jQuery object

Renamed it from `element` to `selector` in https://github.com/iFixit/react-commerce/pull/381/commits/f71029468b10af5103d2eb5646945d04cfe69fcf.